python/lsst/ts/salobj/sal_info.py

```diff
@@ -153,7 +158,12 @@ def __init__(self, domain, name, index=0):
         if not self.idl_loc.is_file():
             raise RuntimeError(f"Cannot find IDL file {self.idl_loc} for name={self.name!r}")
         self.parse_idl()
-        self.ackcmd_type = ddsutil.get_dds_classes_from_idl(self.idl_loc, f"{self.name}::ackcmd")
+        if self.index != 0 and not self.indexed:
```

Comment:
self.indexed is set by parse_idl, so it certainly exists at this point (but just barely). I'm reluctant to set it any earlier (e.g. in the constructor) since I would have to set it to an incorrect value at that point.

I don't understand how making it a class attribute (shared between instances) would be appropriate. It is intrinsically specific to a given instance.

Reply:
But we're inside `__init__` here, aren't we? Oh, I see: it gets set by parse_idl on the line above this. Hmm... Maybe put a comment there to make it clear that that value gets set in that call?